src/Chat.tsx

```diff
@@ -86,6 +92,34 @@ export class Chat extends React.Component<ChatProps, {}> {
         });
     }
 
+    private handleKeyDownCapture(evt: React.KeyboardEvent<HTMLDivElement>) {
+        const target = evt.target as HTMLElement;
+        const tabIndex = getTabIndex(target);
+
+        if (
+            target === findDOMNode(this.chatviewPanel)
+            || typeof tabIndex !== 'number'
+            || tabIndex < 0
+        ) {
+            evt.stopPropagation();
+
+            let key: string;
+
+            // Quirks: onKeyDown we re-focus, but the newly focused element does not receive the subsequent onKeyPress event
+            //         It is working in Chrome/Firefox/IE, confirmed not working in Edge/16
+            //         So we are manually appending the key if they can be inputted in the box
+            if (/(^|\s)Edge\/16\./.test(navigator.userAgent)) {
```

Comment:
are we going to revisit this in Edge 17?

Reply:
Yes. I have file a bug to Edge team.